code/game/objects/effects/contraband.dm

```diff
@@ -223,23 +223,22 @@
 
 	to_chat(user, span_notice("You start placing the poster on the wall...") )
 
-	var/obj/structure/sign/poster/D = P.poster_structure
+	var/obj/structure/sign/poster/placed_poster = P.poster_structure
 
-	var/temp_loc = get_turf(user)
-	flick("poster_being_set",D)
-	D.forceMove(src) //deletion of the poster is handled in poster/Exited(), so don't have to worry about P anymore.
-	playsound(D.loc, 'sound/items/poster_being_created.ogg', 100, TRUE)
+	flick("poster_being_set", placed_poster)
+	placed_poster.forceMove(src) //deletion of the poster is handled in poster/Exited(), so don't have to worry about P anymore.
+	playsound(src, 'sound/items/poster_being_created.ogg', 100, TRUE)
 
-	if(do_after(user, PLACE_SPEED, target=src))
-		if(!D || QDELETED(D))
-			return
+	var/atom/user_drop_location = user.drop_location() //cache this so it just falls to the ground if they move.
+	if(!do_after(user, PLACE_SPEED, placed_poster, extra_checks = CALLBACK(placed_poster, /obj/structure/sign/poster.proc/snowflake_wall_turf_check, src)))
```

Comment:
```suggestion
	if(!do_after(user, PLACE_SPEED, placed_poster, extra_checks = CALLBACK(placed_poster, .proc/istype, src, /turf/closed/wall)))
```

Let me know if this works.

Reply:
I spent almost an hour trying to figure that one out last night before opening the PR, doesn't work sadly.

Reply:
Dang, kind of surprised the internal byond procs can't be used like that.  I know I asked on discord months ago about doing this for macros but Mothblocks advised me not to do it that way.